src/main/cc/wfa/virtual_people/common/field_filter/equal_filter_test.cc

```diff
@@ -0,0 +1,249 @@
+          int32_value: 2
+        }
+      }
+  )PROTO", &test_proto_2));
+  EXPECT_FALSE(field_filter->IsMatch(test_proto_2));
+}
+
+TEST(EqualFilterTest, TestInt64) {
+  std::unique_ptr<FieldFilter> field_filter = FilterFromProtoText(R"PROTO(
+      name: "a.b.int64_value"
+      op: EQUAL
+      value: "1"
+  )PROTO");
+
+  TestProto test_proto_1, test_proto_2;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(R"PROTO(
+      a {
+        b {
+          int64_value: 1
+        }
+      }
+  )PROTO", &test_proto_1));
+  EXPECT_TRUE(field_filter->IsMatch(test_proto_1));
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(R"PROTO(
+      a {
+        b {
+          int64_value: 2
+        }
+      }
+  )PROTO", &test_proto_2));
+  EXPECT_FALSE(field_filter->IsMatch(test_proto_2));
+}
+
+TEST(EqualFilterTest, TestUInt32) {
+  std::unique_ptr<FieldFilter> field_filter = FilterFromProtoText(R"PROTO(
+      name: "a.b.uint32_value"
+      op: EQUAL
+      value: "1"
+  )PROTO");
+
+  TestProto test_proto_1, test_proto_2;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(R"PROTO(
+      a {
+        b {
+          uint32_value: 1
+        }
+      }
+  )PROTO", &test_proto_1));
+  EXPECT_TRUE(field_filter->IsMatch(test_proto_1));
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(R"PROTO(
+      a {
+        b {
+          uint32_value: 2
+        }
+      }
+  )PROTO", &test_proto_2));
+  EXPECT_FALSE(field_filter->IsMatch(test_proto_2));
+}
+
+TEST(EqualFilterTest, TestUInt64) {
+  std::unique_ptr<FieldFilter> field_filter = FilterFromProtoText(R"PROTO(
+      name: "a.b.uint64_value"
+      op: EQUAL
+      value: "1"
+  )PROTO");
+
+  TestProto test_proto_1, test_proto_2;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(R"PROTO(
+      a {
+        b {
+          uint64_value: 1
+        }
+      }
+  )PROTO", &test_proto_1));
+  EXPECT_TRUE(field_filter->IsMatch(test_proto_1));
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(R"PROTO(
+      a {
+        b {
+          uint64_value: 2
+        }
+      }
+  )PROTO", &test_proto_2));
+  EXPECT_FALSE(field_filter->IsMatch(test_proto_2));
+}
+
+TEST(EqualFilterTest, TestBool) {
+  std::unique_ptr<FieldFilter> field_filter = FilterFromProtoText(R"PROTO(
+      name: "a.b.bool_value"
+      op: EQUAL
+      value: "true"
+  )PROTO");
+
+  TestProto test_proto_1, test_proto_2;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(R"PROTO(
+      a {
+        b {
+          bool_value: true
+        }
+      }
+  )PROTO", &test_proto_1));
+  EXPECT_TRUE(field_filter->IsMatch(test_proto_1));
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(R"PROTO(
+      a {
+        b {
+          bool_value: false
+        }
+      }
+  )PROTO", &test_proto_2));
+  EXPECT_FALSE(field_filter->IsMatch(test_proto_2));
+}
+
+TEST(EqualFilterTest, TestEnum) {
+  std::unique_ptr<FieldFilter> field_filter = FilterFromProtoText(R"PROTO(
```

Comment:
@efoxepstein - what do you think of all these raw uses of proto? Would it be better for there to be helper functions to build these in code?

Reply:
I'm happy with the way it is here. I find the textprotos easier to read than code.